### PR TITLE
Add custom mocha reporter

### PR DESCRIPTION
**Pull request by sandersky**
### This project uses [semver](semver.org), please check the scope of this pr:
 - [ ] #patch# - backwards-compatible bug fix
 - [x] #minor# - adding functionality in a backwards-compatible manner
 - [ ] #major# - incompatible API change

# CHANGELOG

* **Added** custom mocha reporter which can be used by consumers.

### CODE REVIEW

File: testem.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+var Reporter = require('./reporter')
+
+module.exports = {
+  framework: 'mocha',
+  'test_page": 'tests/index.html?hidepassed&coverage',
+  'disable_watching': true,
+  'launch_in_ci': [
```
Comment (sophypal) [inline testem.js]: What's with the red here.

File: reporter.js [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+}
+
+/**
+ * Sort tests by duration
+ * @param {TestResult} a - first test
+ * @param {TestResult} b - second test
+ * @returns {Number} number indicating sort order
+ */
+function testSorter (a, b) {
+  if (a.runDuration < b.runDuration) {
```
Comment (sophypal) [inline reporter.js]: I usually do `return a.runDuration - b.runDuration` for ordinal sorts. You can actually return ranges > 1 or < 1.

File: testem.js (hunk 1 above)
Comment (sandersky) [reply testem.js]: I converted from a `.json` file to a `.js` file. Not sure why Github is highlighting things in red.

**Comment by sophypal** [comment]
👍 

[Approved with PullApprove](https://pullapprove.com/ciena-blueplanet/ember-test-utils/pull-request/8/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ciena-blueplanet/ember-test-utils)

**Pull request merged**